Missing dependency history

The `CoreAdminContext` uses `history` package but this package is not required on the `package.json`. 

This leads to this error when I run my react app `npm run start`:
```
/node_modules/ra-core/esm/core/CoreAdminContext.js
Module not found: Can't resolve 'history' in '/Users/aidrissi/apps/myapp/node_modules/ra-core/esm/core'
```

When I check why this package is needed I get:
```
npm ls ra-core
app@0.1.0 /Users/me/apps/app
`-- @api-platform/admin@2.7.2
  `-- react-admin@3.19.6
    +-- ra-core@3.19.6
    +-- ra-i18n-polyglot@3.19.6
    | `-- ra-core@3.19.6  deduped
    `-- ra-language-english@3.19.6
      `-- ra-core@3.19.6  deduped
```

--- a/packages/ra-core/package.json
+++ b/packages/ra-core/package.json
@@ -38,7 +38,6 @@
         "connected-react-router": "^6.5.2",
         "cross-env": "^5.2.0",
         "final-form": "^4.20.4",
-        "history": "^4.7.2",
         "ignore-styles": "~5.0.1",
         "ra-test": "^3.19.10",
         "react": "^17.0.0",
@@ -55,6 +54,7 @@
     "peerDependencies": {
         "connected-react-router": "^6.5.2",
         "final-form": "^4.20.2",
+        "history": "^4.7.2",
         "react": "^16.9.0 || ^17.0.0",
         "react-dom": "^16.9.0 || ^17.0.0",
         "react-final-form": "^6.5.2",

--- a/packages/ra-core/package.json
+++ b/packages/ra-core/package.json
@@ -38,6 +38,7 @@
         "connected-react-router": "^6.5.2",
         "cross-env": "^5.2.0",
         "final-form": "^4.20.4",
+        "history": "^4.7.2",
         "ignore-styles": "~5.0.1",
         "ra-test": "^3.19.10",
         "react": "^17.0.0",

